### ISSUE TITLE
SseEventSource does not get the close call from server after upgrading to Quarkus 2.1.0.Final or later

### ISSUE THREAD

**Issue by johan-hanson** [quarkusio/quarkus#21558]
### Describe the bug

I have created an application that uses a SseEventSource to consume events from a service. The service in some cases close the connection to the client. Using Quarkus 2.0.3.Final or earlier builds the client gets the disconnected and auto connect. So far so good. Upgrading to Quarkus 2.1.0.Final or later the client does not get disconnected. The client just stops receiving events. SseEventSource::isOpen is still true

### Expected behavior

Client should get the close call from server and auto connect

### Actual behavior

Client still think the connection is open but no events are consumed.

### How to Reproduce?

[sse-example.zip](https://github.com/quarkusio/quarkus/files/7568462/sse-example.zip)

The example inlcudes two projects. One for the broker and on for the subscriber.

1. Run in a command prompt or terminal **gradlew build** in _path to project/sse-broker_
2. Run **gradlew quarkusDev** in _path to project/sse-broker_
3. Run in another command prompt **gradlew build** in _path to project/sse-test_
4. Run **gradlew quarkusDev** in _path to project/sse-test_
5. Observe that the log _Number of events: x, connected true_ still counts after the connection is closed (after a minute)
6. Update Quarkus version in _path to project/sse-test_/gradle.properties to **2.1.0.Final** and restart the applications
7. Now observe that the log _Number of events: x, connected true_ stops counting after the connections is closed

### Output of `java -version`

11

### Quarkus version or git rev

2.1.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 7.1

**Comment by gsmet** [comment]
@stuartwdouglas do you happen to know who is responsible for this part?

**Comment by stuartwdouglas** [comment]
@jimma This is caused by https://github.com/resteasy/resteasy/commit/c364a3e3bfc725131d59311d35fef92362b8fd83#diff-3dd8163eff5a9534a4c0f7f98982ed905b4b5591766889e44749b4d524e19551

Basically now instead of reconnecting it will just kinda hang.

**Comment by stuartwdouglas** [comment]
Looks like there is a follow up commit here: https://github.com/resteasy/resteasy/commit/85c23adf513fee62862bc6e4000ff0168bc4c69b#diff-3dd8163eff5a9534a4c0f7f98982ed905b4b5591766889e44749b4d524e19551

This does not take the `alwaysReconnect` param value into account, it seems like it should either reconnect or close based on this value?

**Comment by jimma** [comment]
> Looks like there is a follow up commit here: [resteasy/resteasy@85c23ad#diff-3dd8163eff5a9534a4c0f7f98982ed905b4b5591766889e44749b4d524e19551](https://github.com/resteasy/resteasy/commit/85c23adf513fee62862bc6e4000ff0168bc4c69b#diff-3dd8163eff5a9534a4c0f7f98982ed905b4b5591766889e44749b4d524e19551)
> 
> This does not take the `alwaysReconnect` param value into account, it seems like it should either reconnect or close based on this value?

@stuartwdouglas  It looks SseEventSource handles based on this value to either reconnect or close the SseEventSource.  To disable reconnect, it needs set this value through SourceBuilder : 
```
      SourceBuilder builder = (SourceBuilder) SseEventSource.target(target);
      SseEventSource sourceImpl = builder.alwaysReconnect(false).build();
```
@johan-hanson Can you please try to set this flag to `SourceBuilder` ?

**Comment by stuartwdouglas** [comment]
@jimma reconnect is expected, the issue is that it no longer works after this commit, as the reconnect code was removed. Now the connection simply hangs, reporting that it is still open but no longer receiving events.

**Comment by jimma** [comment]
@stuartwdouglas  @johan-hanson  Get it.  I need to fix this with adding reconnect or close here:
https://github.com/resteasy/resteasy/pull/2770/files#diff-3dd8163eff5a9534a4c0f7f98982ed905b4b5591766889e44749b4d524e19551R381.
The Resteasy issue is created : https://issues.redhat.com/browse/RESTEASY-3051
I already fixed this locally and will merge to upstream soon.

**Comment by gsmet** [comment]
https://issues.redhat.com/browse/RESTEASY-3051 has been fixed in RESTEasy 4.7.4.Final so this issue should be fixed starting with Quarkus 2.6.0.CR1.
